rest_framework/mixins.py

```diff
@@ -69,8 +71,10 @@ def update(self, request, *args, **kwargs):
 
         if getattr(instance, '_prefetched_objects_cache', None):
             # If 'prefetch_related' has been applied to a queryset, we need to
-            # forcibly invalidate the prefetch cache on the instance.
+            # forcibly invalidate the prefetch cache on the instance,
+            # and then re-prefetch related objects
             instance._prefetched_objects_cache = {}
+            prefetch_related_objects([instance], *self.get_prefetch_related())
```

Comment:
Given that we are already relying on a private attribute for `_prefetched_objects_cache`, we may want to also rely on [`QuerySet._prefetch_related_lookups`](https://github.com/django/django/blob/ca9872905559026af82000e46cde6f7dedc897b6/django/db/models/query.py#L1101) for getting the `prefetch_related()` set on the queryset.

This would allow this to be implemented in a way that also fixes the issue for existing queries without requiring the `prefetch_related` to be defined multiple times for a queryset. The downside would be that it would require the query to be statically defined on the view which might not be compatible many of the more advanced use cases that tend to require prefetching.

Reply:
Hi @kevin-brown thanks very much for your comment and sorry for my late reply.

Using `_prefetch_related_lookups` is really a good idea and I've updated as you suggested.